File: frontend/src/api/ai.ts
import { api } from './client'
import type { EntryKind, PaidBy, PayMethod } from './entries'

export interface AiDraft {
  kind: EntryKind
  name: string
  quantity: string | null
  unit: string | null
  /** JAMI summa */
  amount: string | null
  unit_price: string | null
  paid_by: PaidBy | null
  payment_method: PayMethod | null
  matched_price_item_id: number | null
  vendor: string | null
  entry_date: string | null
  note: string | null
  confidence: 'high' | 'low'
}

export const aiParse = (projectId: number, text: string) =>
  api<AiDraft>(`/api/projects/${projectId}/ai-parse`, {
    method: 'POST',
    body: JSON.stringify({ text }),
  })

export function receiptScan(projectId: number, image: Blob) {
  const fd = new FormData()
  fd.append('file', image, 'receipt.jpg')
  return api<AiDraft>(`/api/projects/${projectId}/receipt-scan`, {
    method: 'POST',
    body: fd,
  })
}

File: backend/app/schemas/ai.py
"""AI kiritish sxemalari — matn va chek uchun yagona javob shakli."""

from datetime import date
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, Field


class AiParseRequest(BaseModel):
    text: str = Field(min_length=1, max_length=2000)


class AiDraft(BaseModel):
    """Tasdiqlash formasi shu javobdan oldindan to'ldiriladi."""

    kind: Literal["work", "material", "expense"] = "material"
    name: str = ""
    quantity: Decimal | None = None
    unit: str | None = None
    # amount — JAMI summa; unit_price = amount / quantity
    amount: Decimal | None = None
    unit_price: Decimal | None = None
    paid_by: Literal["master", "client"] | None = None
    payment_method: Literal["cash", "card", "transfer"] | None = None
    matched_price_item_id: int | None = None
    vendor: str | None = None
    entry_date: date | None = None
    note: str | None = None
    confidence: Literal["high", "low"] = "low"
